src/components/qas/questions/qaCalculators/sizeCalculator/index.jsx

```diff
@@ -0,0 +1,242 @@
+    const { value } = ev.currentTarget;
+    this.handleBlurFocusEvent(value, 'p', 'focus');
+  };
+
+  handleBlurFocusEvent = (value, index, eventType) => {
+    // TODO: Toggle expander onExpanderClick() to close
+    const { question, answerHandler } = this.props;
+    const { id } = question;
+    const { value: oldValue } = this.state;
+    const newVal = {
+      ...oldValue,
+      [index]: value,
+    };
+    answerHandler(id, newVal || ' ', eventType);
+
+    this.setState(prevState => ({
+      ...prevState,
+      [index]: newVal,
+      hasFocus: eventType === 'focus',
+    }));
+  };
+
+  render() {
+    const { question, answer, activeId } = this.props;
+    const {
+      // cardActive,
+      hasFocus,
+      answerable,
+    } = this.state;
+
+    const { data } = answer || {};
+    const { h: absoluteMagnitude, p: albedo, d: diameter } = data || {};
+
+    const { id, questionType, label, placeholder } = question;
+    const isTextArea = questionType === 'textArea';
+    const active = activeId === id;
+    const answered = !isEmpty(answer);
+    const cardClasses = classnames('qa-card', {
+      active: hasFocus /* || cardActive */,
+    });
+    const fieldClasses = classnames('qa-text-input', {
+      answered,
+      unanswered: !answered,
+      answerable: answerable || answered || active,
+    });
+
+    return (
+      <Card
+        className={cardClasses}
+        expanded={hasFocus}
+        onExpanderClick={() => {}}
+      >
+        <CardText>
+          <div className="col-50">
+            <TextField
+              id={`text-${isTextArea ? 'area' : 'input'}-${id}`}
+              className={fieldClasses}
+              type="number"
+              min="0"
+              leftIcon={<HIcon />}
+              label={<div dangerouslySetInnerHTML={renderDef(label)} />}
+              lineDirection="center"
+              placeholder={placeholder}
+              defaultValue={answered ? absoluteMagnitude : null}
+              onBlur={this.handleHBlur}
+              onFocus={this.handleHFocus}
+              onChange={this.handleHChange}
+              disabled={!(answerable || answered || active)}
+            />
+          </div>
+          <div className="col-50">
+            <TextField
+              id={`text-${isTextArea ? 'area' : 'input'}-${id}`}
+              className={fieldClasses}
+              type="number"
+              min="0"
+              leftIcon={<PIcon />}
+              label={<div dangerouslySetInnerHTML={renderDef(label)} />}
+              lineDirection="center"
+              placeholder={placeholder}
+              defaultValue={answered ? albedo : null}
+              onBlur={this.handlePBlur}
+              onFocus={this.handlePFocus}
+              onChange={this.handlePChange}
+              disabled={!(answerable || answered || active)}
+            />
+          </div>
+          <div className="results-list">
+            <CalculatedMeasurement unit="D" value={diameter} />
```

Comment:
`D` always `= ?` on pageload even when there is an answer present.  Probably just related to the issue described above in the `componentDidMount` method.